### PR TITLE
Add half-value Munsell chips beyond 2.5

### PR DESCRIPTION
**Pull request by dylanbeaudette**
Starting from #318, I've added all half-value Munsell chips to the `munsell` LUT used in aqp. This works well, but breaks existing tests, and many years of expected behavior. A result of `10YR 3.5/4` isn't always the best answer, even though it might be the closest conversion from e.g. sRGB coordinates or the result of a simulated mixture.

At this point, I think that it makes sense to just interpolate 2.5, 8.5, 9.5 value chips + spectra to minimize complexity in related functions and current behavior. A more intensive addition to the LUT and reference spectra should be another issue/PR.

TODO:
  - [ ] ~~Think about new arguments to color conversion functions the limit results to "standard" value chips~~
  - [x] finish interpolation of half-value spectra
  - [x] update all tests, many will fail at first
  - [x] more tests on expanded `munsell` LUT
  - [x] update documentation
  - [x] re-run closest munsell chips code -> update associated .rda

Also worth including:
  - [x] update rounding logic in `munsell2rgb()`
  - [ ] ~~implement optimization in `munsell2rgb()`~~
  - [x] misc. TODOs in `munsell2rgb()`
  - [x] double-check `getClosestMunsellChip()` behavior

### CODE REVIEW

**Comment by dylanbeaudette** [comment]
Interpolation of Munsell reference spectra results in some odd outliers, at the edges of reference spectra. Will deal with that in a different issue.

**Comment by dylanbeaudette** [comment]
I think this is just about done, now with a smaller scope.

@brown-ag at your convenience, can you please try re-building the closest munsell chips data and update the associated test? Those tests are currently skipped. Thanks.

**Comment by brownag** [comment]
Should be done in https://github.com/ncss-tech/aqp/pull/319/commits/f24b51bd3d6a12788c8c48ea4af8f1d06bcec2ac

**Comment by dylanbeaudette** [comment]
I've noticed some subtle differences in simulated mixtures of Munsell colors:

Before: 2.5GY 5/1
image


After: 2.5GY 4/2
{E89B6F3B-0385-4963-B905-02C7929F9C1A}

File: misc/utils/Munsell/local-functions.R [hunk 1]
```diff
@@ -38,8 +38,47 @@ interpolateChroma <- function(m.i) {
...
+interpolateValue2 <- function(m.i) {
+  
+  # can only proceed with >=2 rows
+  # some combinations of hue, value, chroma have 1 row.
+  # there will be other combinations created by split() with 0 rows
+  if(nrow(m.i) < 2) {
+    return(NULL)
+  }
+  
+  # linear interpolation ~ munsell value
```
Comment (brownag) [inline misc/utils/Munsell/local-functions.R]: Update comment

**Comment by brownag** [comment]
> I've noticed some subtle differences in simulated mixtures of Munsell colors:

I am not sure I completely understand why adding 8.5 and 9.5 value chips would change this result. Is this because of the use of `splinefun()` for interpolating values v.s. former linear interpolation via `approxfun()`?

**Comment by dylanbeaudette** [comment]
Adding the 8.5 and 9.5 chroma entries to the `munsell` LUT and reference spectra required a re-write of multiple interpolation steps. In doing so I updated:
  * converted all interpolation to `splinefun`, vs. use of `approxfun` in the interpolation of a single, new 2.5 value chip/spectra
  * new interpolation of value/chroma/spectra now use the entire set of available points vs. just those bracketing the "missing" points → this leads to a more stable curve
  * added a check/fix for spectra reflectance < 0, caused by spline interpolation overshoots at a few extremes. This requires a little more testing as part of a new issue to be created. Previously this was only done for chroma, I added a similar check/fix for value
 
In short, the "addition" of 8.5 and 9.5 value chips / spectra triggered a cascade of updates and enhancements.

**Comment by dylanbeaudette** [comment]
OK. I think that we are good to go here.
